packages/backend/src/sagas/incomingTypingSaga.js

```diff
@@ -0,0 +1,32 @@
+import {
+  call,
+  put,
+  takeLatest
+} from 'redux-saga/effects';
+
+import deleteActivity from '../Actions/deleteActivity';
+import { UPSERT_ACTIVITY } from '../Actions/upsertActivity';
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function isTypingActivity({
+  type,
+  payload
+}) {
+  if (type === UPSERT_ACTIVITY) {
+    const { activity: { type } } = payload;
+
+    return type === 'typing';
+  }
+}
+
+export default function* () {
+  yield takeLatest(isTypingActivity, function* ({ payload: { activity } }) {
+    const id = activity.id;
+
+    yield call(sleep, 5000);
```

Comment:
Is this deleting the typing activity after 5 seconds... should the bot, instead, send an update to the conversation when it wants the typing to go away?

Reply:
I would love to see both, but unfortunately, the current protocol will only send typing and no stop typing.